adagrams/game.py

```diff
@@ -1,11 +1,181 @@
+        'S', 'S', 'S', 'S', 
+        'T', 'T', 'T', 'T', 'T', 'T', 
+        'U', 'U', 'U', 'U', 
+        'V', 'V', 
+        'W', 'W', 
+        'X', 
+        'Y', 'Y', 
+        'Z')
+    
+    draw_letters = []
+    
+    
+    for letter in letter_pool_list:
+        while len(draw_letters) < 10:
+            tile = random.choice((letter_pool_list))
+            if draw_letters.count(tile) < letter_pool_list.count(tile):
+                draw_letters.append(tile)
+        
+        
+        return draw_letters
+
+    
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    words = word.upper()
+
+    for letters in (words):
+        if (all(letters in letter_bank for letters in words)):
+            while words.count(letters) <= letter_bank.count(letters):
+                return True
+            
+    else:
+            return False
+
+
+
+
 
 def score_word(word):
-    pass
+    #need to create a score dictionary
+    # have a score counter that loops through letter in word and adds the value from the dictionary to the score counter
+    score_chart = {
+        'A':1,
+        'E':1,
+        'I':1,
+        'O':1,
+        'U':1,
+        'L':1,
+        'N':1,
+        'R':1,
+        'S':1,
+        'T':1,
+        'D':2,
+        'G':2,
+        'B':3,
+        'C':3,
+        'M':3,
+        'P':3,
+        'F':4,
+        'H':4,
+        'V':4,
+        'W':4,
+        'Y':4,
+        'K':5,
+        'J':8,
+        'X':8,
+        'Q':10,
+        'Z':10
+    }
+
+    sum_score = 0
+    words = word.upper()
+    for letter in words:
+        sum_score += score_chart[letter]
+    if len(words) in range(7,11):
+        sum_score += 8
+        
+
+    return sum_score
+        
+
+    
+    
+    
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    
+
+    word_list_dict = {}
+    max_word_list = []
+    word_len_10_list = []
+    
+
+    for word in word_list:
+        word_list_dict[word] = score_word(word)
+        print(word_list_dict)
+        highest_word_score = max(word_list_dict.values())
+    for key, val in word_list_dict.items():
+        if word_list_dict[key] == highest_word_score:
+            max_word_list.append(key)
+
+    if len(max_word_list) > 1:
+        if len(max(max_word_list, key = len)) == 10:
+
+            for word in max_word_list:
+                if len(word) == 10:
+                    word_len_10_list.append(word)
+                    if len(word_len_10_list) > 1:
+                        return tuple([word_len_10_list[0], highest_word_score])
+                    else:
+                        return tuple([word_len_10_list[0], highest_word_score])
+
+        else:
+            return tuple([(min(max_word_list, key = len)), highest_word_score])
+    else:
+        return tuple([max(word_list_dict), highest_word_score ])
+
+                    
+
+                
+
+    
+    return word_list_dict
+
+
+
+
+
+
+    
```

Comment:
Phalesa, you did really good on this project! I hope you get more comfortable talking about your code, because I would love for you to share with the other students! Please feel free to reach out if you have any questions about the comments I left! Again, great job! 🤩